pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -223,6 +226,13 @@ func (s *Reconciler) Delete(ctx context.Context) error {
 		return errors.Wrapf(err, "Unable to delete network interface")
 	}
 
+	// Create public IP resource if requested
```

Comment:
Reading these few lines I see:

- Create (comment)
- Generate
- Delete

How does this work? Is this to generate based on a template which then must be discarded?

Reply:
So the assumption is if `PublicIP` is set, public IP resource is create for each such vm in Azure. Here, we provide only a name for the resource. The rest of the azure library takes care of providing a template that is used to create the resource. On deletion, we just delete by the resource name.